backend_plugin/src/main/evaluators/BlockCommentEvaluator.java

```diff
@@ -3,114 +3,118 @@
- * then the user should be notified of the block comment feature in Eclipse.
- */
 public class BlockCommentEvaluator extends FeatureEvaluator {
+
+	private int lastCommentedLine;
+	private long lastCommentedLineTimeStamp;
 		
-	private boolean firstDoubleBackSlashDetected;
-	private IRegion prevRegion;
-
-	// position of the previous document change
-	private int prevOffset;
-	// text of the previous document change
-	private String prevInsert;
-
-	// position of the current document change
-	private int currentOffset;
-	// text of the current document change
-	private String currentInsert;
-
-	private static final String SINGLE_SLASH = "/";
-	private static final String DOUBLE_SLASH = "//";
-	
 	/**
 	 * Default constructor
 	 */
 	public BlockCommentEvaluator() {
 		this.featureID = "blockCommentSuggestion";
-		firstDoubleBackSlashDetected = false;
+		this.lastCommentedLine = -2;
 	}
-	
+
 	/**
-	 * Keeps track of DocumentEvent changes and determines of the user comments out multiple sequential lines of code. 
-	 * @param event the change detected by the DocumentChange Listener
-	 * @return true if the user comments two sequential lines of code, false otherwise
+	 * Evaluates whether a warning should be triggered about the block comment feature
 	 */
 	public boolean evaluateDocumentChanges(DocumentEvent event) {
-		if (doubleBackSlash(event)) {
-			IDocument doc = event.getDocument();
-			try {
-				// Verify that the double slash is at the beginning of the line
-				// grab the position of the start of the line where the document change was
-				int lineOffset = doc.getLineInformationOfOffset(currentOffset).getOffset();
-				String textBeforeSlashes = doc.get(lineOffset, prevOffset-lineOffset);
-
-				textBeforeSlashes.trim();
-				if (textBeforeSlashes.isEmpty()) {
-
-					// If we have two back slashes, then check if we've found two previously
-					if (firstDoubleBackSlashDetected) {
-						IRegion currentRegion = event.getDocument().getLineInformationOfOffset(currentOffset);
-
-						// Check if consecutive lines
-						int prevToCurrentDif = currentRegion.getOffset() - (prevRegion.getOffset() + prevRegion.getLength());
-						int currentToPrevDif = prevRegion.getOffset() - (currentRegion.getOffset() + currentRegion.getLength());
-
-						// update the prevRegion
-						prevRegion = currentRegion;
-
-						// Uses a buffer of 2 characters because sometimes IRegion doesn't count the newly inserted "//"
-						if ((prevToCurrentDif >= 0 && prevToCurrentDif <= 2) ||
-								(currentToPrevDif >= 0 && currentToPrevDif <= 2)) {
-							return true;
-						}
-
-						// If we haven't seen a double back slash yet
-					} else {
-						firstDoubleBackSlashDetected = true;
-						prevRegion = event.getDocument().getLineInformationOfOffset(currentOffset);
-					}
+		try {
+
+			// Get the document the user is typing in
+			IDocument document = event.getDocument();
```

Comment:
Since we are getting the same document every event (I think?), we should store the document as a member variable rather than extract it from the `DocumentEvent` every time. This also seems like something other feature evaluators might want, so it should maybe live in the `FeatureEvaluator` abstract class.

Reply:
I've updated the PR to place the document as a field of the FeatureEvaluator abstract class.